compiler/passes/errorHandling.cpp

```diff
@@ -131,29 +131,29 @@ class ErrorHandlingVisitor : public AstVisitorTraverse {
 public:
   ErrorHandlingVisitor       (ArgSymbol* _outFormal, LabelSymbol* _epilogue);
 
-  virtual bool enterTryStmt  (TryStmt*   node);
-  virtual void exitTryStmt   (TryStmt*   node);
-  virtual bool enterCatchStmt(CatchStmt*   node);
-  virtual void exitCatchStmt (CatchStmt*   node);
-  virtual bool enterCallExpr (CallExpr*  node);
+  virtual bool enterTryStmt (TryStmt*   node);
+  virtual void exitTryStmt  (TryStmt*   node);
+  virtual void exitCatchStmt(CatchStmt* node);
+  virtual bool enterCallExpr(CallExpr*  node);
 
 private:
   struct TryInfo {
     VarSymbol*   errorVar;
     LabelSymbol* handlerLabel;
+    TryStmt*     tryStmt;
+    BlockStmt*   tryBody;
   };
 
   std::stack<TryInfo> tryStack;
+  std::stack<TryInfo> catchesStack;
   ArgSymbol*          outError;
   LabelSymbol*        epilogue;
-  bool                insideCatch;
 
-  AList     lowerCatches      (TryStmt*   tryStmt,  VarSymbol* errorVar,
-                               TryInfo*   outerTry);
-  AList     setOutGotoEpilogue(VarSymbol* error);
-  AList     errorCond         (VarSymbol* errorVar, BlockStmt* thenBlock,
-                               BlockStmt* elseBlock = NULL);
-  CallExpr* haltExpr          ();
+  void   lowerCatches      (TryInfo    info);
```

Comment:
I'd expect you to pass a (const) reference to TryInfo for this function, vs the current copy.